test/CollectiveGovernance.t.sol

```diff
@@ -75,276 +79,371 @@ contract CollectiveGovernanceTest is Test {
+        _storage.setQuorumThreshold(PROPOSAL_ID, 2, supervisor);
+        _storage.makeReady(PROPOSAL_ID, supervisor);
         vm.stopPrank();
+        vm.prank(supervisor);
+        governance.openVote(PROPOSAL_ID);
         vm.prank(voter1);
         governance.voteFor(PROPOSAL_ID);
         assertEq(_storage.quorum(PROPOSAL_ID), 1);
     }
 
-    function testFailCastOneVoteWithBurnedClass() public {
-        address strategy = address(governance);
-        vm.startPrank(strategy);
-        _storage.registerVoterClassOpenVote(PROPOSAL_ID);
-        _storage.burnVoterClass(PROPOSAL_ID);
-        _storage.setQuorumThreshold(PROPOSAL_ID, 2);
-        _storage.makeReady(PROPOSAL_ID);
-        governance.openVote(PROPOSAL_ID);
+    function testCastOneVoteWithBurnedClass() public {
+        vm.startPrank(_cognate);
+        _storage.registerVoterClassOpenVote(PROPOSAL_ID, supervisor);
+        _storage.burnVoterClass(PROPOSAL_ID, supervisor);
+        _storage.setQuorumThreshold(PROPOSAL_ID, 2, supervisor);
+        _storage.makeReady(PROPOSAL_ID, supervisor);
         vm.stopPrank();
+        vm.prank(supervisor);
+        governance.openVote(PROPOSAL_ID);
+        vm.expectRevert("Not a voter");
         vm.prank(voter1);
         governance.voteFor(PROPOSAL_ID);
     }
 
-    function testFailCastVoteNotOpened() public {
-        vm.prank(supervisor);
-        _storage.registerVoter(PROPOSAL_ID, voter1);
+    function testCastVoteNotOpened() public {
+        vm.prank(_cognate);
+        _storage.registerVoterClassVoterPool(PROPOSAL_ID, supervisor);
+        vm.prank(_cognate);
+        _storage.registerVoter(PROPOSAL_ID, voter1, supervisor);
+        vm.expectRevert("Voting is closed");
         vm.prank(voter1);
         governance.voteFor(PROPOSAL_ID);
     }
 
-    function testFailCastTwoVote() public {
-        vm.startPrank(supervisor);
-        _storage.registerVoter(PROPOSAL_ID, voter1);
-        _storage.setQuorumThreshold(PROPOSAL_ID, 2);
-        governance.openVote(PROPOSAL_ID);
+    function testCastTwoVote() public {
+        vm.startPrank(_cognate);
+        _storage.registerVoterClassVoterPool(PROPOSAL_ID, supervisor);
+        _storage.registerVoter(PROPOSAL_ID, voter1, supervisor);
+        _storage.setQuorumThreshold(PROPOSAL_ID, 2, supervisor);
+        _storage.makeReady(PROPOSAL_ID, supervisor);
         vm.stopPrank();
+        vm.prank(supervisor);
+        governance.openVote(PROPOSAL_ID);
         vm.prank(voter1);
         governance.voteFor(PROPOSAL_ID);
+        vm.expectRevert("Already voted");
         vm.prank(voter1);
         governance.voteFor(PROPOSAL_ID);
     }
 
-    function testFailUndoRequiresOpen() public {
-        vm.startPrank(supervisor);
-        _storage.enableUndoVote(PROPOSAL_ID);
-        _storage.registerVoter(PROPOSAL_ID, voter1);
-        _storage.setQuorumThreshold(PROPOSAL_ID, 2);
-        _storage.makeReady(PROPOSAL_ID);
+    function testCastDoubleVoteOnTransferToken() public {
```

Comment:
@dannyadkins Here is the test that covers voting from a transferred token on a separate address